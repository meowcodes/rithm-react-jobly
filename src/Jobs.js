import React, { Component } from 'react';
import JoblyApi from './JoblyAPI';
import Search from './Search';
import JobCard from './JobCard';

/**
 * Recieves a list of job data from API request & search keyword from Search
 * Renders all/searched jobs (JobCard) and search bar (Search)
 * Sends individual job data to JobCard
 */

class Jobs extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
      jobData: [],
      error: null
    }
    this.updateQuery = this.updateQuery.bind(this)
  }

  // get job data from API
  async componentDidMount() {
    try {
      let data = await JoblyApi.getJobs()

      this.setState({
        jobData: data,
        loading: false
      });
    } catch(err) {
      this.setState({
        error: err
      });
    }
  }

  // updates query make another API req
  async updateQuery(query) {
    try {
      let searchData = await JoblyApi.getJobs({"search": query})
      this.setState({
        jobData: searchData
      });
    } catch(err) {
      this.setState({
        error: err
      });
    }
  }


  render() {
    const jobs = this.state.jobData.map( job => 
      <JobCard title={job.title} salary={job.salary} equity={job.equity} key={ job.id }/>
    )
    return (
      <div className="jobs">
        { !this.state.error && this.state.loading
          ? <p>Loading...</p>
          : <div>
            <Search triggerUpdateQuery={this.updateQuery} />
            { jobs }
          </div>
        } 
        { this.state.error && <p>Something happened</p>}
      </div>
    );
  }
}

export default Jobs;
