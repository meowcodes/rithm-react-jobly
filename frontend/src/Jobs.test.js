import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { MemoryRouter } from 'react-router-dom';
import Jobs from './Jobs';


// full mount
it('mounts without crashing', function() {
  mount(
    <MemoryRouter>
      <Jobs 
        currJobs={[]}
        triggerApply={()=> null}/>
    </MemoryRouter>
  );
});

// snapshot test
it('matches snapshot', function() {
  let wrapper = shallow(<Jobs 
                            currJobs={[]}
                            triggerApply={()=> null}/>);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
