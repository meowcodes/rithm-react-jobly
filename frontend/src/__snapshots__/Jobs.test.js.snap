// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="jobs"
>
  <p>
    Loading...
  </p>
</div>
`;
